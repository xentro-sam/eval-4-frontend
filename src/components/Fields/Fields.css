#field-body {
    color: rgb(79, 79, 79);
    font-size: 1vw;
    font-weight: 600;
    border-radius: 5px;
    padding: 1vw 2vw;
    background-color: white;
    margin: 1vw;
    display: flex;
    justify-content: space-between;
}

#field-type {
    color: rgb(190, 190, 190);
}

#field-delete > img {
    width: 1vw;
    height: 1vw;
    cursor: pointer;
}