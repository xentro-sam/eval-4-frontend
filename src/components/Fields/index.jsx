import * as React from 'react';
import './Fields.css';
import PropTypes from 'prop-types';
import deleteIcon from '../../assets/trash-delete-recycle-bin-bucket-waste@3x.png';
import editIcon from '../../assets/user-pencil-write-ui-education@3x.png';
import makeRequest from '../../utils/makeRequest';
import {UPDATE_CONTENT_TYPE as updateContentType, UPDATE_CONTENT_FIELD_NAME as updateContentFieldName} from '../../constants/apiEndPoints';

export default function Fields(props) {
  const [showEdit, setShowEdit] = React.useState(false);
  const handleDelete = async () => {
    await makeRequest(updateContentType(props.contentTypeId), {
      data: {
        contentTypeFields: [props.fieldName],
        operation: 'remove',
      },
    });
    props.removeField(props.fieldName);
  };
  const toggleEdit = () => {
    setShowEdit(!showEdit);
  };
  const handleEditName = () => {
    const newTitle = document.getElementById('field-name-edit-input').value;
    if (newTitle) {
      const data = {
        oldName: props.fieldName,
        newName: newTitle,
      };
      makeRequest(updateContentFieldName(props.contentTypeId), {
        data,
      }).then((res) => {
        props.renameFields(props.fieldName, newTitle);
        setShowEdit(false);
      });
    }
  };

  return (
    <div id="field-body">
      <div id="field-name">
        {props.fieldName}
        {showEdit && (
          <div>
            <input type="text" id="field-name-edit-input" placeholder="Enter new name" />
            <button id="field-name-edit-button" onClick={handleEditName}>Save</button>
          </div>
        )}
      </div>
      <div id="field-type">{props.fieldType}</div>
      <div id="field-actions">
        <div id="field-delete">
          <img src={deleteIcon} alt="delete-icon" onClick={handleDelete} />
        </div>
        <div id="field-edit">
          <img src={editIcon} alt="edit-icon" onClick={toggleEdit} />
        </div>
      </div>
    </div>
  );
}

Fields.propTypes = {
  fieldName: PropTypes.string.isRequired,
  fieldType: PropTypes.string.isRequired,
  contentTypeId: PropTypes.number.isRequired,
  removeField: PropTypes.func.isRequired,
  renameFields: PropTypes.func.isRequired,
};
