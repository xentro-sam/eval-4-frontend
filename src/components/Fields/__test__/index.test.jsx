import React from 'react';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Fields from '../index';
import {waitFor} from '@testing-library/dom';
import makeRequest from '../../../utils/makeRequest';
import {act} from 'react-dom/test-utils';

jest.mock('../../../utils/makeRequest');

const mockData = {
  fieldName: 'name',
  fieldType: 'text',
  contentTypeId: 1,
  removeField: jest.fn(),
  renameFields: jest.fn(),
};

describe('ContentType', () => {
  it('should render correctly', async () => {
    expect(makeRequest).not.toHaveBeenCalled();
    const {asFragment} = await act( async () => render(<Fields {...mockData} />));
    await waitFor(() => {
      expect(screen.getByText('text')).toBeTruthy();
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
