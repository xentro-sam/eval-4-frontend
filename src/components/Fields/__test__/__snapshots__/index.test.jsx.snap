// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentType should render correctly 1`] = `
<DocumentFragment>
  <div
    id="field-body"
  >
    <div
      id="field-name"
    >
      name
    </div>
    <div
      id="field-type"
    >
      text
    </div>
    <div
      id="field-actions"
    >
      <div
        id="field-delete"
      >
        <img
          alt="delete-icon"
          src="trash-delete-recycle-bin-bucket-waste@3x.png"
        />
      </div>
      <div
        id="field-edit"
      >
        <img
          alt="edit-icon"
          src="user-pencil-write-ui-education@3x.png"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
